pkg/daemon/controller/controller.go

```diff
@@ -271,9 +270,15 @@ func (sc *ServiceController) Reconcile(_ context.Context, req ctrl.Request) (res
 	if err = sc.reconcileServices(cmData.Services); err != nil {
 		return ctrl.Result{}, err
 	}
-	if err = sc.reconcileEnvironmentVariables(cmData.EnvironmentVars, &node); err != nil {
+	restartRequired, err := sc.reconcileEnvironmentVariables(cmData.EnvironmentVars)
```

Comment:
Do we really need a bool here? Don't we always need to restart if no error is returned?

Reply:
No we don't always have to restart, we only restart when we change a value in the OS registry.
On each reconcile, we do a comparison between the proxy variable value currently set in the Windows OS registry with the value we have specified in the configmap.